libyul/AsmAnalysis.cpp

```diff
@@ -45,22 +45,20 @@ using namespace solidity::langutil;
 
 bool AsmAnalyzer::analyze(Block const& _block)
 {
-	m_success = true;
+	auto watcher = m_errorReporter.errorWatcher();
```

Comment:
```suggestion
	auto noNewError = m_errorReporter.errorWatcher();
```
Without a more specific naming, it is hard to remember if it returns "newError" or "noNewError".

In a previous version, you called this `ok`, which would also be fine, I think.

Reply:
We could also invert the meaning and make the error watcher return true on new error - which would maybe be more in line with the name of the class.

Reply:
I guess with such naming `operator bool()` would make more sense than `bool operator()()`.

Reply:
> We could also invert the meaning and make the error watcher return true on new error

I would rather keep the default meaning, "true is OK, all good, continue normally". Alternatively we could have explicit `watcher.success()` and (for convenience) `watcher.error()`.

Reply:
I second the idea of being explicit. `watcher.ok()` is good in my opinion because it actually reads like a condition. Something like`!watcher.hasErrors()` or `watcher.empty()` would be even better.  I never really liked using non-booleans as conditions. It may be shorter but it's awful to read.

Reply:
That sounds good! I'm fine with things like `watcher.hasErrors`, although it would be good to have an indication that this only concerns new errors.

Reply:
Oh, naming... I think `watcher` does not _have_ errors. It rather sees/spots them. `watcher.errorOccurred()` or just `watcher.nok()` maybe?

Reply:
`newError()`?

Reply:
That looks rather as an error factory. `recentError()`? (it also sounds weird though.)